File: Resources/CourseraAlgorithmicToolbox/week3_greedy_algorithms/4_maximum_advertisement_revenue/dot_product.hs
-- by Kirill Elagin

max_dot_product :: [Int] -> [Int] -> Integer
max_dot_product as bs = result -- write your code here

main :: IO ()
main = do
  _  <- getLine
  as <- fmap (map read . words) getLine
  bs <- fmap (map read . words) getLine
  print $ max_dot_product as bs

File: Resources/CourseraAlgorithmicToolbox/week3_greedy_algorithms/4_maximum_advertisement_revenue/DotProduct.kt
import java.util.*

fun maxDotProduct(a: IntArray, b: IntArray): Long {
    //write your code here
    var result: Long = 0
    for (i in a.indices) {
        result += (a[i] * b[i]).toLong()
    }
    return result
}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    val a = IntArray(n)
    for (i in 0 until n) {
        a[i] = scanner.nextInt()
    }
    val b = IntArray(n)
    for (i in 0 until n) {
        b[i] = scanner.nextInt()
    }
    println(maxDotProduct(a, b))
}